### PR TITLE
Disc 637 adjust open api from config

### PR DESCRIPTION
**Pull request by VictorHarbo**
Opening as draft as there are multiple decisions to be made before this can be implemented correctly. 

Experimenting with injecting yaml into POM to then use it in OpenAPI specs. 

See internal issue for context

### CODE REVIEW

**Comment by VictorHarbo** [comment]
PR updated with working plugin. However the plugin haven't been released yet, therefor a manual installation is required for testing it currently.

**Comment by VictorHarbo** [comment]
Plugin has now been released. Ready for official review.

File: pom.xml [hunk 1]
```diff
@@ -81,6 +81,13 @@
             </snapshots>
         </repository>
     </repositories>
+    <pluginRepositories>
+        <pluginRepository>
+            <id>comphenix-rep</id>
+            <name>Comphenix Maven Releases</name>
+            <url>http://repo.comphenix.net/content/groups/public</url>
+        </pluginRepository>
+    </pluginRepositories>
```
Comment (tokee) [inline pom.xml:90]: Seems like leftovers from experimenting?

File: pom.xml [hunk 2]
```diff
@@ -331,7 +338,35 @@
...
+                            <yamlType>List</yamlType>
+                            <yamlPath>allowed_origins</yamlPath>
+                            <sourceKey>name</sourceKey>
+                            <destinationKey>allowed_origins</destinationKey>
+                            <createEnum>true</createEnum>
+                        </YamlResolver>
+                        <YamlResolver>
+                            <yamlType>Single-value</yamlType>
+                            <yamlPath>allowed_origins[0].name</yamlPath>
+                            <destinationKey>example_origin</destinationKey>
```
Comment (tokee) [inline pom.xml:365]: Very nice with the example!

File: pom.xml [hunk 3]
```diff
@@ -331,7 +338,35 @@
...
+                    <yamlResolvers>
+                        <YamlResolver>
+                            <yamlType>List</yamlType>
+                            <yamlPath>allowed_origins</yamlPath>
+                            <sourceKey>name</sourceKey>
+                            <destinationKey>allowed_origins</destinationKey>
+                            <createEnum>true</createEnum>
+                        </YamlResolver>
+                        <YamlResolver>
+                            <yamlType>Single-value</yamlType>
```
Comment (tokee) [inline pom.xml:363]: Running
```shell
mvn clean package -Dskiptests
```
gives me
```
[ERROR] Failed to execute goal dk.kb.util:injection-maven-plugin:1.0.0:read-yaml-properties (default)
  on project ds-storage: Unable to parse configuration of mojo dk.kb.util:injection-maven-plugin:1.0.0:
  read-yaml-properties for parameter sourceKey: Cannot find 'sourceKey' in class dk.kb.util.YamlResolver
 -> [Help 1]
```

**Review by tokee — CHANGES_REQUESTED** [review]
Looks very elegant. Only problem is, it does not run :-)

File: pom.xml (hunk 3 above)
Comment (VictorHarbo) [reply pom.xml:363]: Doh. I've been to fast. Sorry!

Comment (VictorHarbo) [reply pom.xml:363]: My guess is that you've installed the plugin manually earlier. Could you please remove the 1.0.0 version installed locally in your .m2 repo and then re-run the command?

**Review by tokee — APPROVED** [review]
Looks good now. And a very nice functionality to have.

**Pull request merged**